core/services/pipeline/task.divide.go

```diff
@@ -0,0 +1,50 @@
+package pipeline
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+)
+
+type DivideTask struct {
+	BaseTask  `mapstructure:",squash"`
+	Input     string `json:"input"`
+	Divisor   string `json:"divisor"`
+	Precision string `json:"precision"`
+}
+
+var _ Task = (*DivideTask)(nil)
+
+func (t *DivideTask) Type() TaskType {
+	return TaskTypeDivide
+}
+
+func (t *DivideTask) Run(_ context.Context, vars Vars, inputs []Result) (result Result) {
+	_, err := CheckInputs(inputs, -1, -1, 0)
+	if err != nil {
+		return Result{Error: errors.Wrap(err, "task inputs")}
+	}
+
+	var (
+		a              DecimalParam
+		b              DecimalParam
+		maybePrecision MaybeInt32Param
+	)
+	err = multierr.Combine(
+		errors.Wrap(ResolveParam(&a, From(VarExpr(t.Input, vars), Input(inputs, 0))), "input"),
+		errors.Wrap(ResolveParam(&b, From(VarExpr(t.Divisor, vars), NonemptyString(t.Divisor))), "divisor"),
+		errors.Wrap(ResolveParam(&maybePrecision, From(VarExpr(t.Precision, vars), t.Precision)), "precision"),
+	)
+	if err != nil {
+		return Result{Error: err}
+	}
+
+	value := a.Decimal().Div(b.Decimal())
+
+	if precision, isSet := maybePrecision.Int32(); isSet {
+		value = value.Round(precision)
```

Comment:
ah this would quietly round twice because decimal.Div default to rounding to 16 precision points, which could be a problem if you specify > 16. I think this would be cleaner:
```
precision, isSet := maybePrecision.Int32()
if isSet {
      value = a.Decimal.DivRound(b.Decimal(), precision)
} else {
      // Note that decimal library defaults to rounding to 16 precision
      //https://github.com/shopspring/decimal/blob/2568a29459476f824f35433dfbef158d6ad8618c/decimal.go#L44
      value = a.Decimal.Div(b.Decimal())
}
```
Alternatively just forcing a precision specification in the spec might be even better